File: python/rules/purpose-and-accuracy/no-speculative-platform-branching.py
import sys
import os
import platform

def get_config_dir() -> str:
    # ruleid: python-no-speculative-platform-branching
    if sys.platform == "win32":
        return r"C:\Config"
    return "/etc/config"

def check_system():
    # ruleid: python-no-speculative-platform-branching
    if platform.system() == "Windows":
        return "win"
    
    # ruleid: python-no-speculative-platform-branching
    if os.name == "nt":
        return "win"

def get_standard_dir() -> str:
    # ok: python-no-speculative-platform-branching
    return "/etc/config"

File: csharp/rules/purpose-and-accuracy/no-speculative-platform-branching.cs
using System;
using System.Runtime.InteropServices;

namespace Thruput.Semgrep.Tests
{
    public class PlatformCheckTests
    {
        public string GetAppPath()
        {
            // ruleid: csharp-no-speculative-platform-branching
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                return @"C:\App";
            }
            return "/var/app";
        }

        public void CheckModernOS()
        {
            // ruleid: csharp-no-speculative-platform-branching
            if (OperatingSystem.IsWindows())
            {
                Console.WriteLine("Windows");
            }
        }

        public string GetStandardAppPath()
        {
            // ok: csharp-no-speculative-platform-branching
            return "/var/app/data";
        }
    }
}
